src/test/java/com/fryrank/dal/UserMetadataDALTests.java

```diff
@@ -0,0 +1,78 @@
+import com.fryrank.model.GetUserMetadataOutput;
+import com.fryrank.model.UserMetadata;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.mongodb.core.FindAndReplaceOptions;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.List;
+
+import static com.fryrank.Constants.ACCOUNT_ID_KEY;
+import static com.fryrank.TestConstants.TEST_ACCOUNT_ID;
+import static com.fryrank.TestConstants.TEST_USER_METADATA_1;
+import static com.fryrank.TestConstants.TEST_USER_METADATA_LIST;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserMetadataDALTests {
+    @Mock
+    MongoTemplate mongoTemplate;
+
+    @InjectMocks
+    UserMetadataDALImpl userMetadataDAL;
+
+    @Test
+    public void testGetUserMetadataForAccountId_happyPath() throws Exception {
+        final Query query = new Query();
+        query.addCriteria(where(ACCOUNT_ID_KEY).is(TEST_ACCOUNT_ID));
+
+        when(mongoTemplate.find(query, UserMetadata.class)).thenReturn(TEST_USER_METADATA_LIST);
+
+        final GetUserMetadataOutput expectedOutput = new GetUserMetadataOutput(TEST_USER_METADATA_LIST);
+        final GetUserMetadataOutput actualOutput = userMetadataDAL.getUserMetadataForAccountId(TEST_ACCOUNT_ID);
+        assertEquals(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testGetUserMetadataForAccountId_noUserMetadata() throws Exception {
+        final Query query = new Query();
+        query.addCriteria(where(ACCOUNT_ID_KEY).is(TEST_ACCOUNT_ID));
+
+        final List<UserMetadata> expectedUserMetadata = List.of();
+        when(mongoTemplate.find(query, UserMetadata.class)).thenReturn(expectedUserMetadata);
+
+        final GetUserMetadataOutput expectedOutput = new GetUserMetadataOutput(expectedUserMetadata);
+        final GetUserMetadataOutput actualOutput = userMetadataDAL.getUserMetadataForAccountId(TEST_ACCOUNT_ID);
+        assertEquals(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testGetUserMetadataForAccountId_nullAccountId() {
+        final Query query = new Query();
+        query.addCriteria(where(ACCOUNT_ID_KEY).is(null));
+
+        final List<UserMetadata> expectedUserMetadata = List.of();
+
+        final GetUserMetadataOutput expectedOutput = new GetUserMetadataOutput(expectedUserMetadata);
+        final GetUserMetadataOutput actualOutput = userMetadataDAL.getUserMetadataForAccountId(TEST_ACCOUNT_ID);
+        assertEquals(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testAddNewReview() throws Exception {
```

Comment:
testUpsertUserMetadata

Reply:
Done in [e281f82](https://github.com/NickPriv/FryRankBackend/pull/75/commits/e281f825a8834d7e490af094d3f5f49d6a0f9de1).